### ISSUE TITLE
bug: Does not produce valid SARIF

### ISSUE THREAD

**Issue by tbutler-qontigo** [terraform-linters/tflint#1657]
### Summary

The SARIF format files produced are not valid according to the SARIF Validator at https://www.sarif.info/Validation

To Reproduce
1. Install the windows version of tflint v0.44.1
2. create a folder `tflint-fail` and a sub-folder `sub`
3. In the subfolder create a file `test.tf` which contains just this line: `variable "test" {}`
4. From the `tflint-fail` folder, run `tflint --format sarif --recursive`
5. Load the output into https://www.sarif.info/Validation
6. See these errors are reported:
* [SARIF2005](http://docs.oasis-open.org/sarif/sarif/v2.1.0/sarif-v2.1.0.html): runs[0].tool.driver: The tool 'tflint' does not provide any of the version-related properties 'version', 'semanticVersion', 'dottedQuadFileVersion'. Providing version information enables the log file consumer to determine whether the file was produced by an up to date version, and to avoid accidentally comparing log files produced by different tool versions.
* [SARIF1007](http://docs.oasis-open.org/sarif/sarif/v2.1.0/sarif-v2.1.0.html): runs[0].results[0].locations[0].physicalLocation.region: This 'region' object does not specify 'startLine', 'charOffset', or 'byteOffset'. As a result, it is impossible to determine whether this 'region' object describes a line/column text region, a character offset/length text region, or a binary region.
* [JSON1008](https://www.sarif.info/undefined): runs[0].results[0].locations[0].physicalLocation.region.startLine: 0 is less than the minimum permitted value of 1.
* [JSON1008](https://www.sarif.info/undefined): runs[0].results[0].locations[0].physicalLocation.region.startColumn: 0 is less than the minimum permitted value of 1.
* [SARIF1002](http://docs.oasis-open.org/sarif/sarif/v2.1.0/sarif-v2.1.0.html): runs[0].results[1].locations[0].physicalLocation.artifactLocation.uri: The string 'sub\test.tf' is not a valid URI reference. URIs must conform to [RFC 3986](https://tools.ietf.org/html/rfc3986).
* [SARIF2005](http://docs.oasis-open.org/sarif/sarif/v2.1.0/sarif-v2.1.0.html): runs[1].tool.driver: The tool 'tflint-errors' does not provide any of the version-related properties 'version', 'semanticVersion', 'dottedQuadFileVersion'. Providing version information enables the log file consumer to determine whether the file was produced by an up to date version, and to avoid accidentally comparing log files produced by different tool versions.

**Expected behavior**
The SARIF produced passes all *mandatory* validation rules - whilst some issues like the lack of version info are mandatory, the SARIF1002 requirement for a valid URI is mandatory.  This affects the ability to import the output and process it in tools like SonarQube

Attached zip contains both the input and the output files
[tflint-fail.zip](https://github.com/terraform-linters/tflint/files/10517057/tflint-fail.zip)

### Command

tflint --format sarif --recursive

### Terraform Configuration

```terraform
As above, literally just a file in a subfolder containing:
variable "test" {}
```

### TFLint Configuration

```HCL
No config file used
```

### Output

```Shell
{
  "version": "2.1.0",
  "$schema": "https://json.schemastore.org/sarif-2.1.0-rtm.5.json",
  "runs": [
    {
      "tool": {
        "driver": {
          "name": "tflint",
          "informationUri": "https://github.com/terraform-linters/tflint",
          "rules": [
            {
              "id": "terraform_required_version",
              "shortDescription": {
                "text": ""
              },
              "helpUri": "https://github.com/terraform-linters/tflint-ruleset-terraform/blob/v0.2.2/docs/rules/terraform_required_version.md"
            },
            {
              "id": "terraform_typed_variables",
              "shortDescription": {
                "text": ""
              },
              "helpUri": "https://github.com/terraform-linters/tflint-ruleset-terraform/blob/v0.2.2/docs/rules/terraform_typed_variables.md"
            },
            {
              "id": "terraform_unused_declarations",
              "shortDescription": {
                "text": ""
              },
              "helpUri": "https://github.com/terraform-linters/tflint-ruleset-terraform/blob/v0.2.2/docs/rules/terraform_unused_declarations.md"
            }
          ]
        }
      },
      "results": [
        {
          "ruleId": "terraform_required_version",
          "level": "warning",
          "message": {
            "text": "terraform \"required_version\" attribute is required"
          },
          "locations": [
            {
              "physicalLocation": {
                "artifactLocation": {
                  "uri": ""
                },
                "region": {
                  "startLine": 0,
                  "startColumn": 0,
                  "endLine": 1,
                  "endColumn": 1
                }
              }
            }
          ]
        },
        {
          "ruleId": "terraform_typed_variables",
          "message": {
          },
            {
                "artifactLocation": {
                  "uri": "sub\\test.tf"
                },
                "region": {
                  "startLine": 1,
                  "startColumn": 1,
                  "endLine": 1,
                  "endColumn": 16
                }
              }
            }
          ]
        {
          "ruleId": "terraform_unused_declarations",
          "level": "warning",
          "message": {
            "text": "variable \"test\" is declared but not used"
          },
          "locations": [
            {
              "physicalLocation": {
                "artifactLocation": {
                  "uri": "sub\\test.tf"
                },
                "region": {
                  "startLine": 1,
                  "startColumn": 1,
                  "endLine": 1,
                  "endColumn": 16
                }
              }
            }
          ]
        }
      ]
    },
    {
      "tool": {
        "driver": {
          "name": "tflint-errors",
          "informationUri": "https://github.com/terraform-linters/tflint"
        }
      },
      "results": []
    }
  ]
}
```

### TFLint Version

0.44.1

### Terraform Version

1.3.7

### Operating System

- [ ] Linux
- [ ] macOS
- [X] Windows

**Comment by wata727** [comment]
Makes sense. Patches are welcome.
